velox/functions/sparksql/specialforms/FromJson.cpp

```diff
@@ -192,10 +193,27 @@ struct ExtractJsonTypeImpl {
         const folly::
             F14FastMap<int64_t, JsonRowSchemaInfo>& /*jsonRowSchemaInfo*/,
         column_index_t /*nodeIndex*/) {
+      if (writer.type()->isShortDecimal()) {
+        return castJsonToDecimal<int64_t>(value, writer);
+      }
       return castJsonToInt<int64_t>(value, writer);
     }
   };
 
+  template <typename Dummy>
+  struct KindDispatcher<TypeKind::HUGEINT, Dummy> {
+    static simdjson::error_code apply(
+        Input value,
+        exec::GenericWriter& writer,
+        bool /*isRoot*/,
+        const folly::
+            F14FastMap<int64_t, JsonRowSchemaInfo>& /*jsonRowSchemaInfo*/,
+        column_index_t /*nodeIndex*/) {
+      VELOX_CHECK(writer.type()->isLongDecimal());
```

Comment:
Spark does not have the hugeint type. I suppose `VELOX_DCHECK` would be good enough.